src/synthesized_insight/metrics/metrics_usage.py

```diff
@@ -1,68 +1,80 @@
-
 from itertools import permutations
 from typing import Optional, Tuple, Union
 
 import numpy as np
 import pandas as pd
 
-from .base import TwoColumnMetric
-
+from .base import DataFrameMetric, OneColumnMetric, TwoColumnMetric, TwoDataFrameMetric
 
-class TwoColumnMap:
-    """Compares columns with the same name from two given dataframes and return a DataFrame
-    with index as the column name and the columns as metric_val and metric_pval(if applicable)"""
 
-    def __init__(self, metric: TwoColumnMetric):
-        self.metric = metric
+class OneColumnMap(DataFrameMetric):
+    def __init__(self, metric: OneColumnMetric):
+        self._metric = metric
         self.name = f'{metric.name}_map'
 
-    def __call__(self, df_old: pd.DataFrame, df_new: pd.DataFrame) -> pd.DataFrame:
-        columns_map = {col: self.metric(df_old[col], df_new[col]) for col in df_old.columns}
-
+    def __call__(self, df: pd.DataFrame) -> pd.DataFrame:
+        columns_map = {col: self._metric(df[col]) for col in df.columns}
         result = pd.DataFrame(
             data=columns_map.values(),
-            index=df_old.columns,
+            index=df.columns,
             columns=['metric_val']
         )
 
-        result.name = self.metric.name
+        result.name = self._metric.name
         return result
 
 
-class CorrMatrix:
+class CorrMatrix(DataFrameMetric):
     """Computes the correlation between each pair of columns in the given dataframe
     and returns the result in a dataframe"""
 
     def __init__(self, metric: TwoColumnMetric):
-        self.metric = metric
+        self._metric = metric
         self.name = f'{metric.name}_matrix'
 
-    def __call__(self, df: pd.DataFrame) -> Tuple[pd.DataFrame, Optional[pd.DataFrame]]:
+    def __call__(self, df: pd.DataFrame) -> Union[pd.DataFrame]:
```

Comment:
Don't think Union is needed here? Union is like "or" so here it's effectively saying "pd.DataFrame or"... with no 2nd option.

Reply:
Unless it's the expected return type of this call somewhere else...